recipes/mysql-connector-cpp/all/conanfile.py

```diff
@@ -0,0 +1,171 @@
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake, CMakeDeps, cmake_layout
+from conan.tools.files import get, replace_in_file, rm, collect_libs, copy
+from conan.tools.scm import Version
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.build import check_min_cppstd
+from conan.tools.env import VirtualBuildEnv
+import os
+
+requird_conan_version = ">=1.54.0"
+
+class MysqlConnectorCPPRecipe(ConanFile):
+    name = "mysql-connector-cpp"
+    license = "GPL-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://dev.mysql.com/doc/connector-cpp/8.1/en/"
+    description = "A MySQL database connector for C++ applications that connect to MySQL servers"
+    topics = ("mysql", "connector", "libmysqlclient", "jdbc")
+    package_type = "library"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False], "with_jdbc": [True, False]}
+    default_options = {"shared": False, "fPIC": True, "with_jdbc": True}
+
+    @property
+    def _min_cppstd(self):
+        return "17"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
+            "gcc": "9" if Version(self.version) >= "8.0.27" else "5.3",
+            "clang": "6",
+        }
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def build_requirements(self):
+        self.tool_requires("protobuf/3.21.12")
+
+    def requirements(self):
+        self.requires("zlib/[>=1.2.10 <2]")
+        self.requires("lz4/1.9.4")
+        self.requires("zstd/1.5.5")
+        self.requires("protobuf/3.21.12")
+        self.requires("boost/1.83.0")
```

Comment:
```suggestion
        self.requires("boost/1.84.0")
```

We have Boost 1.84.0 available already: https://conan.io/center/recipes/boost?version=1.84.0